platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/stats/ReconnectMapMetrics.java

```diff
@@ -0,0 +1,167 @@
+     *     in addition to emitting metrics for this object, may be null.
+     */
+    public ReconnectMapMetrics(
+            @NonNull final Metrics metrics,
+            @Nullable final String label,
+            @Nullable final ReconnectMapStats aggregateStats) {
+        Objects.requireNonNull(metrics, "metrics must not be null");
+        this.aggregateStats = aggregateStats;
+
+        this.transfersFromTeacher = metrics.getOrCreate(
+                new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("transfersFromTeacher", label))
+                        .withDescription("number of transfers from teacher to learner"));
+        this.transfersFromLearner = metrics.getOrCreate(
+                new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("transfersFromLearner", label))
+                        .withDescription("number of transfers from learner to teacher"));
+
+        this.internalHash =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalHash", label))
+                        .withDescription("number of internal node hashes transferred"));
+        this.internalCleanHash =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalCleanHash", label))
+                        .withDescription("number of clean internal node hashes transferred"));
+        this.internalData =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalData", label))
+                        .withDescription("number of internal node data transferred"));
+        this.internalCleanData =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalCleanData", label))
+                        .withDescription("number of clean internal node data transferred"));
+
+        this.leafHash = metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafHash", label))
+                .withDescription("number of leaf node hashes transferred"));
+        this.leafCleanHash =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafCleanHash", label))
+                        .withDescription("number of clean leaf node hashes transferred"));
+        this.leafData = metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafData", label))
+                .withDescription("number of leaf node data transferred"));
+        this.leafCleanData =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafCleanData", label))
+                        .withDescription("number of clean leaf node data transferred"));
+    }
+
+    private static String formatName(final String name, final String label) {
+        return (label == null || label.isBlank() ? name : (name + "_" + label + "_")) + "Total";
```

Comment:
`_Total` looks redundant here. In grafana, metric names are used to get the list of virtual map names dynamically, rather than to have a hard-coded list. It is done using a regex like

```
/virtual_map_vmap_size_(.*)/
```

Something similar may be needed for these new metrics. The pattern may be

```
/internalHashes_(.*)_Total/
```

but for consistency I'd propose to get rid of `Total` altogether.

Reply:
It's added for a reason. In a previous PR where I introduced new metrics, I was told that they should always end with a units suffix:

https://github.com/hashgraph/hedera-services/pull/12759#discussion_r1565867375

So we have to have the "Total" suffix here.

I'm keeping this intact until/unless you get @beeradb to comment here and suggest otherwise.